src/Common/AzureRest/AzureRestOperations.cs

```diff
@@ -276,26 +297,26 @@ public List<T> GetResouceList<T>(string resourceUri, string apiVersion, IDiction
             return BeginHttpGetMessagesAsync<List<T>>(resourceUri, apiVersion, customHeaders, cancellationToken).GetAwaiter().GetResult();
         }
 
-        public IPage<T> GetResoucePage<T>(string resourceUri, string apiVersion, IDictionary<string, IList<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public P GetResoucePage<P, T>(string resourceUri, string apiVersion, IDictionary<string, IList<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken)) where P : IPage<T>
```

Comment:
could we just use IPage<T> GetResourcePage<T>?

Reply:
As discussed, I need to pass implementation into this method for JSON deserialization